frontend/public/components/utils/dropdown.jsx

```diff
@@ -62,17 +57,24 @@ export class DropdownMixin extends React.PureComponent {
     });
   }
 
-  toggle () {
-    this.setState({active: !this.state.active});
+  toggle (e) {
+    if (this.state.active) {
+      this.hide(e);
+    } else {
+      this.show(e);
+    }
   }
 
   show (e) {
     e && e.stopPropagation();
+    window.removeEventListener('click', this.listener); // Just in case
```

Comment:
A comment explaining this would be great.

Reply:
Fixed.